### DISCUSSION TITLE
Minimum system requirement to run RuFFle

### DISCUSSION THREAD

**Discussion by ROBERT-MCDOWELL** [ruffle-rs/ruffle#829] [General]
I wonder if it's possible to have JS script or else
to check if the wasm engine version is compatible with Ruffle so
the developer can redirect to a legacy or modern website pages?
Thanks

**Comment by SoftCreatR** [comment]
I don't think, that the engine version matters. Every browser that supports WASM (see https://caniuse.com/wasm) should be able to run ruffle. To simply check for WASM support, this may help:

```js
const supported = (() => {
    try {
        if (typeof WebAssembly === "object"
            && typeof WebAssembly.instantiate === "function") {
            const module = new WebAssembly.Module(Uint8Array.of(0x0, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00));
            if (module instanceof WebAssembly.Module)
                return new WebAssembly.Instance(module) instanceof WebAssembly.Instance;
        }
    } catch (e) {
    }
    return false;
})();

console.log(supported ? "WebAssembly is supported" : "WebAssembly is not supported");
```

**Comment by ROBERT-MCDOWELL** [comment]
Thanks for this code, as Wasm specs are still changing I noticed that some browser not updated since 2 or 3 years with wasm compatible are not able to run some recent wasm code.

**Reply by SoftCreatR** [reply]
I'm aware of that. If you want to check for specific WASM features, this might help: https://github.com/GoogleChromeLabs/wasm-feature-detect

I think, it's hard to tell, which features are required by ruffle and which aren't. However, I think the dev team focuses on full Chrome & FF support (see https://webassembly.org/roadmap/) and avoid using functionalities, which are not supported by major browsers.

**Comment by ROBERT-MCDOWELL** [comment]
useful links thanks!
